mache/spack/chrysalis_gnu_openmpi.yaml

```diff
@@ -16,16 +16,16 @@ spack:
     unify: true
   packages:
     all:
-      compiler: [gcc@9.2.0]
+      compiler: [gcc@11.2.0]
       providers:
-        mpi: [openmpi@4.1.3]
+        mpi: [openmpi@4.1.6]
 {% if e3sm_lapack %}
-        lapack: [intel-mkl@2020.4.304]
+        lapack: [intel-oneapi-mkl@2022.1.0]
 {% endif %}
     bzip2:
       externals:
       - spec: bzip2@1.0.8
-        prefix: /gpfs/fs1/soft/chrysalis/spack/opt/spack/linux-centos8-x86_64/gcc-9.2.0/bzip2-1.0.8-n5r3ey3
+        prefix: /gpfs/fs1/soft/chrysalis/spack/opt/spack/linux-centos8-x86_64/gcc-11.2.0/bzip2-1.0.8-n5r3ey3
```

Comment:
Listing:
```
/gpfs/fs1/soft/chrysalis/spack/opt/spack/linux-centos8-x86_64/gcc-11.2.0/
```
shows that hashes have changed for these packages, too.  Fortunately, we can get away with using the old versions with the old compiler because these are just compiled tools, not libraries we will link in.  So let's just leave `bzip2`, `curl`, diffutils`, `libiconv, `libxml2`, `xz` and `zlib` as they were.